L/LAMMPS/build_tarballs.jl

```diff
@@ -75,18 +75,19 @@ augment_platform_block = """
 # platforms are passed in on the command line
 # platforms = supported_platforms(; experimental=true)
 platforms = supported_platforms()
-platforms = expand_cxxstring_abis(platforms)
-
 platforms, platform_dependencies = MPI.augment_platforms(platforms)
 # Avoid platforms where the MPI implementation isn't supported
 # OpenMPI
-platforms = filter(p -> !(p["mpi"] == "openmpi" && arch(p) == "armv6l" && libc(p) == "glibc"), platforms)
+platforms = filter(p -> !(p["mpi"] == "openmpi" && nbits(p) == 32), platforms)
```

Comment:
You don't need to filter out 32-bit platforms if you build against OpenMPI 4: https://github.com/JuliaPackaging/Yggdrasil/blob/27db5ba5a624df4809faef80a12297ccbc570ab4/H/HDF5/build_tarballs.jl#L245.  However, one issue I never had the time to look at is #8211 (this compat doesn't seem to be propagated correctly)

Reply:
How does: ` OpenMPI_compat="4.1.6, 5" ` work? Do we not need to build against OpenMPI v4 and OpenMPI v5 separately?

Reply:
the compat forces the lower version as build version, for OpenMPI we don't need separate builds because the two major versions are ABI-compatible